pydis_site/apps/api/viewsets/bot/nomination.py

```diff
@@ -1,32 +1,224 @@
+from collections import ChainMap
+
+from django.utils import timezone
+from django_filters.rest_framework import DjangoFilterBackend
+from rest_framework import status
+from rest_framework.decorators import action
 from rest_framework.exceptions import ValidationError
+from rest_framework.filters import OrderingFilter, SearchFilter
+from rest_framework.mixins import (
+    CreateModelMixin,
+    ListModelMixin,
+    RetrieveModelMixin,
+)
 from rest_framework.response import Response
-from rest_framework.viewsets import ModelViewSet
+from rest_framework.viewsets import GenericViewSet
 
 from pydis_site.apps.api.models.bot import Nomination
 from pydis_site.apps.api.serializers import NominationSerializer
 
 
-class NominationViewSet(ModelViewSet):
-    """View providing CRUD operations on helper nominations done through the bot."""
+class NominationViewSet(CreateModelMixin, RetrieveModelMixin, ListModelMixin, GenericViewSet):
+    """
+    View providing CRUD operations on helper nominations done through the bot.
+
+    ## Routes
+    ### GET /bot/nominations
+    Retrieve all nominations.
+    May be filtered and ordered by the query parameters.
+
+    #### Query parameters
+    - **active** `bool`: whether the nomination is still active
+    - **actor__id** `int`: snowflake of the user who nominated the user
+    - **user__id** `int`: snowflake of the user who received the nomination
+    - **ordering** `str`: comma-separated sequence of fields to order the returned results
+
+    Invalid query parameters are ignored.
+
+    #### Response format
+    >>> [
+    ...     {
+    ...         'id': 1,
+    ...         'active': false,
+    ...         'actor': 336843820513755157,
+    ...         'reason': 'They know how to explain difficult concepts',
+    ...         'user': 336843820513755157,
+    ...         'inserted_at': '2019-04-25T14:02:37.775587Z',
+    ...         'unnominate_reason': 'They were helpered after a staff-vote',
+    ...         'unwatched_at': '2019-04-26T15:12:22.123587Z'
+    ...     }
+    ... ]
+
+    #### Status codes
+    - 200: returned on success
+
+    ### GET /bot/nominations/<id:int>
+    Retrieve a single nomination by ID.
+
+    ### Response format
+    >>> {
+    ...     'id': 1,
+    ...     'active': true,
+    ...     'actor': 336843820513755157,
+    ...     'reason': 'They know how to explain difficult concepts',
+    ...     'user': 336843820513755157,
+    ...     'inserted_at': '2019-04-25T14:02:37.775587Z',
+    ...     'unnominate_reason': 'They were helpered after a staff-vote',
+    ...     'unwatched_at': '2019-04-26T15:12:22.123587Z'
+    ... }
+
+    ### Status codes
+    - 200: returned on succes
+    - 404: returned if an nomination with the given `id` could not be found
```

Comment:
```suggestion
    - 404: returned if a nomination with the given `id` could not be found
```
unless my english fails me